agent/consul/internal_endpoint.go

```diff
@@ -414,8 +414,15 @@ func (m *Internal) GatewayServices(args *structs.ServiceSpecificRequest, reply *
 				}
 			}
 
+			// We log a warning here to indicate that there is a potential
+			// misconfiguration. We explicitly do NOT return an error because this
+			// can occur in the course of normal operation by deleting a
+			// configuration entry or starting the proxy before registering the
+			// config entry.
 			if !found {
-				return fmt.Errorf("service %q is not a configured terminating-gateway or ingress-gateway", args.ServiceName)
+				m.logger.Warn("service is not a configured terminating-gateway or ingress-gateway",
+					"service", args.ServiceName,
+				)
```

Comment:
Could we make this a bit more explicit?

How about:
```suggestion
				m.logger.Warn("there is no terminating-gateway or ingress-gateway configuration entry associated with this gateway",
					"gateway", args.ServiceName,
				)
```

Reply:
done.